src/mc_control/CMakeLists.txt

```diff
@@ -3,31 +3,56 @@
 #
 
 macro(add_controller controller_name controller_SRC controller_HDR)
-  add_library(${controller_name} SHARED ${controller_SRC} ${controller_HDR})
-  set_target_properties(${controller_name} PROPERTIES FOLDER controllers)
-  set_target_properties(${controller_name} PROPERTIES COMPILE_FLAGS
-    "-DMC_CONTROL_EXPORTS" PREFIX "")
-  target_link_libraries(${controller_name} PUBLIC mc_control)
-  install(TARGETS ${controller_name}
-    ARCHIVE DESTINATION "${MC_CONTROLLER_LIBRARY_INSTALL_PREFIX}$<$<CONFIG:debug>:/debug>"
-    LIBRARY DESTINATION "${MC_CONTROLLER_LIBRARY_INSTALL_PREFIX}$<$<CONFIG:debug>:/debug>"
-    RUNTIME DESTINATION "${MC_CONTROLLER_RUNTIME_INSTALL_PREFIX}$<$<CONFIG:debug>:/debug>"
-  )
+  if(MC_RTC_BUILD_STATIC)
+    target_sources(mc_control PRIVATE ${controller_SRC} ${controller_HDR})
+  else()
+    add_library(${controller_name} SHARED ${controller_SRC} ${controller_HDR})
+    set_target_properties(${controller_name} PROPERTIES FOLDER controllers)
+    set_target_properties(${controller_name} PROPERTIES COMPILE_FLAGS
+      "-DMC_CONTROL_EXPORTS" PREFIX "")
+    target_link_libraries(${controller_name} PUBLIC mc_control)
+    install(TARGETS ${controller_name}
+      ARCHIVE DESTINATION "${MC_CONTROLLER_LIBRARY_INSTALL_PREFIX}$<$<CONFIG:debug>:/debug>"
+      LIBRARY DESTINATION "${MC_CONTROLLER_LIBRARY_INSTALL_PREFIX}$<$<CONFIG:debug>:/debug>"
+      RUNTIME DESTINATION "${MC_CONTROLLER_RUNTIME_INSTALL_PREFIX}$<$<CONFIG:debug>:/debug>"
+    )
```

Comment:
Not directly related to this PR, but recently Tsuru got quite confused because he mistakenly compiled a plugin in both release and debug. At runtime since `mc_rtc` was compiled in release, it was picking the (older) installed release version and not the (newer) debug version. I don't think there is a problem with what's being done here, but it might be worth considering adding a tutorial page on the topic in the future.

Reply:
Agreed, opened #93 on this topic